src/uct/tcp/tcp_ep.c

```diff
@@ -2163,3 +2163,28 @@ uct_tcp_ep_check(uct_ep_h tl_ep, unsigned flags, uct_completion_t *comp)
     hdr->length = 0;
     return uct_tcp_ep_am_send(ep, hdr);
 }
+
+int uct_tcp_ep_is_connected(uct_ep_h tl_ep,
+                            const uct_ep_is_connected_params_t *params)
+{
+    uct_tcp_ep_t *ep = ucs_derived_of(tl_ep, uct_tcp_ep_t);
+    ucs_status_t status;
+    struct sockaddr_storage dest_addr;
+    int is_connected;
+
+    UCT_EP_PARAMS_CHECK_IS_CONNECTED_DEV_IFACE_ADDRS(params);
+
+    status = uct_tcp_ep_set_dest_addr(params->device_addr, params->iface_addr,
+                                      (struct sockaddr*)&dest_addr);
+    if (status != UCS_OK) {
+        return 0;
+    }
+
+    is_connected = !ucs_sockaddr_cmp((struct sockaddr*)&ep->peer_addr,
+                                     (struct sockaddr*)&dest_addr, &status);
+    if (status != UCS_OK) {
```

Comment:
how could status be != UCS_OK here?

Reply:
if ucs_sockaddr_cmp returned UCS_ERR_INVALID_PARAM